### ISSUE TITLE
deprecate unpacking marks in parameter values and provide a clean alternative

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pytest-dev/pytest#1748]
the mechanism we use to makr parameter sets 

like

``` python
import pytest
@pytest.mark.parametrize(("n", "expected"), [
  (1, 2),
  pytest.mark.foo((2, 3)),
  (3, 4),
])
def test_increment(n, expected):
    assert n + 1 == expected
```

means that we have to unpack mark decorator objects in a very error prone/messy way
and in addition we end up with the inability to safely use function that way

i propose replacing it with something like

``` python
import pytest
@pytest.mark.parametrize(("n", "expected"), [
  (1, 2),
  pytest.marked_value((2, 3)), marks=pytest.mark.foo)
  (3, 4),
])
def test_increment(n, expected):
    assert n + 1 == expected
```

**Comment by The-Compiler** [comment]
Assuming you mean `pytest.marked_value((2, 3), marks=pytest.mark.foo)` (without the extra `)`), that sounds like a good idea.

It'd also make using marks with arguments nicer IMHO:

before: `pytest.mark.xfail(reason='...')((1, 2))`
after: `pytest.marked_value((1, 2), marks=pytest.mark.xfail(reason='...'))`

**Comment by RonnyPfannschmidt** [comment]
laos a "fun detail" currently `pytest.mark.xfail(reason='...')((1, 2))` is equivalent to `pytest.mark.xfail((1, 2), reason='...')`

which is just plain horror

**Comment by RonnyPfannschmidt** [comment]
fixed in #1921
